src/components/date/__internal__/date-formats/index.ts

```diff
@@ -140,7 +140,18 @@ interface LocaleFormats {
   format: string;
 }
 
-const getFormatData = ({ code = "en-GB" }: DateFnsLocale): LocaleFormats => {
+const getFormatData = (
+  { code = "en-GB" }: DateFnsLocale,
+  dateFormatOverride?: string
+): LocaleFormats => {
+  if (dateFormatOverride) {
+    const formats = getInputFormatsArrayForLocale(dateFormatOverride);
```

Comment:
comment: this will potentially flip the days and months for that locale. If you see below the input of `05/04` is being taken as the 4th May instead of 5th April.

![image](https://github.com/user-attachments/assets/09998ffa-5ee8-4bd1-8ab8-3f0bdf4e5748)

I think we need to still use the locale to generate the arrays which could very roughly look like this:

```
const getFormatData = (
  { code = "en-GB" }: DateFnsLocale,
  dateFormatOverride?: string
): LocaleFormats => {
  if (dateFormatOverride) {
    const { format } = getOutputFormatForLocale(code);
      let formatFromLocale;

    switch(code) {
      case "en-CA":
      case "en-US":
        formatFromLocale = "MM/dd/yyyy";
        break;
      case "fr-CA":
      case "en-ZA":
      case "ar-EG":
        formatFromLocale = "dd/MM/yyyy";
        break;
      default:
        formatFromLocale = format;
    }
   
    const formatsForLocale = getInputFormatsArrayForLocale(formatFromLocale);

    return {
      format: dateFormatOverride,
      formats: generateFormats(formatsForLocale, "/"),
    };
  }

  if (["en-CA", "en-US"].includes(code)) {
    const format = "MM/dd/yyyy";
    const formats = getInputFormatsArrayForLocale(format);
    return {
      format,
      formats: generateFormats(formats, "/"),
    };
  }

  const { format, separator } = getOutputFormatForLocale(code);
  const outputFormat = ["fr-CA", "en-ZA", "ar-EG"].includes(code)
    ? "dd/MM/yyyy"
    : format;
  const formatsForLocale = getInputFormatsArrayForLocale(outputFormat);

  return {
    format: outputFormat,
    formats: generateFormats(
      formatsForLocale,
      separator,
      getTrailingChar(format)
    ),
  };
};
```

![image](https://github.com/user-attachments/assets/4d2c556a-db97-468a-9c4f-84b97a8a367d)

Reply:
Thanks Ed, I'll make that change and update the tests to cover off those scenarios where the day and month could be swapped